base/inc/AdePT/BenchmarkManager.h

```diff
@@ -0,0 +1,172 @@
+  double getDurationSeconds(TTag tag)
+  {
+    return (float)(std::chrono::duration_cast<PRECISSION>(timers[tag].accumulated_duration)).count()/PRECISSION::period::den;
+  }
+
+  //Sets the accumulated duration in seconds for the specified tag
+  void setDurationSeconds(TTag tag, double duration)
+  {
+    timers[tag].accumulated_duration = std::chrono::duration_cast<CLOCK::duration>(std::chrono::duration<double>(duration));
+  }
+
+  //Adds to the accumulated duration for the specified tag
+  void addDurationSeconds(TTag tag, double duration)
+  {
+    if(timers.find(tag) == timers.end())
+    {
+      //Initialize if the timer didn't exist
+      timers[tag] = TimeInfo{CLOCK::now(), std::chrono::seconds(0)};
+    }
+    timers[tag].accumulated_duration += std::chrono::duration_cast<CLOCK::duration>(std::chrono::duration<double>(duration));
+  }
+
+  //Empties the timer map
+  void reset()
+  {
+    timers.clear();
+  }
+
+  //Removes an timer from the map
+  void removeTimer(TTag tag)
+  {
+    timers.erase(tag);
+  }
+
+  //Checks if an timer is in the map
+  bool hasTimer(TTag tag)
+  {
+    return timers.find(tag) != timers.end();
+  }
+
+  //Export a CSV file with the timer names as labels and the accumulated time for each
+  //If the file is not empty, write only the times
+  void exportCSV(std::string filename="benchmark")
+  {
+  	std::string path = output_dir + "/" + filename + ".csv";
+
+  	//Create the output directory if it doesn't exist.
+  	std::filesystem::create_directory(output_dir);
+
+  	bool first_write = !std::filesystem::exists(path);
+
+		std::ofstream output_file;
+  	output_file.open(path, std::ofstream::app);
+
+  	//Write the header only the first time
+  	if(first_write)
+  	{
+  		for(auto iter = timers.begin(); iter != timers.end(); ++iter)
+  		{
+  			output_file << iter -> first;
+  			if(std::next(iter) != timers.end()) output_file << ", ";
+  		}
+  		output_file << std::endl;
+  	}
+
+  	//Print 6 decimal digits
+  	output_file << std::setprecision(6) << std::fixed;
+    for(auto iter = timers.begin(); iter != timers.end(); ++iter)
+    {
+    	output_file << getDurationSeconds(iter->first);
+      if(std::next(iter) != timers.end()) output_file << ", ";  
+    }
+    output_file << std::endl;
+
+    std::cout << "BENCHMARK: Results saved to benchmark/" << filename << ".csv" << std::endl;
+    
+  }
+
+  static bool isActive()
+  {
+    return true;
+  }
+
+};
+
+#else
+
+template <class TTag> 
+class BenchmarkManager
+{
+	public:
+	  BenchmarkManager(){};
+	  ~BenchmarkManager(){};
+		void timerStart(TTag tag){}
+		void timerEnd(TTag tag){}
+    double getDurationSeconds(TTag tag){return 0;}
+    void addDurationSeconds(TTag tag, double duration){}
+    void reset(){}
+    void removeTimer(TTag tag){}
+    bool hasTimer(TTag tag){return false;}
+		void exportCSV(TTag filename){}
+    static bool isActive(){return false;}
+};
```

Comment:
Why do you need a second implementation if BENCHMARK is not on? IMO the utility should always be the same and the protection should be put in a macro BENCHMARK_START/END(tag) that would compile to empty if BENCHMARK is not on. You can then protect per example if you want calling benchmarking on conditions.